### ISSUE TITLE
Add rule for image without line break

### ISSUE THREAD

**Issue by Marcono1234** [DavidAnson/markdownlint#474]
### Description

Often users want images to be displayed on a separate line. They then add a _single_ line break and then the image reference, e.g.:
```md
some text
[My image](image.png)
```

However, due to Markdown line break handling, this will actually be considered an image which is on the same line as the text. In most cases this renders as desired, but only because Markdown renderer tend to wrap images to the next line if there is not enough space. Viewing the Markdown file on a page with a different width or with a different screen size can suddenly show the image on the same line.

It might therefore be good to flag image references, which follow a non-empty line without trailing 2 spaces (to cause line break).
Though, maybe there are corner cases where such behavior is desired.

### Example
```md
Test:
![Alt text](https://davidanson.gallerycdn.vsassets.io/extensions/davidanson/vscode-markdownlint/0.45.0/1635486276371/Microsoft.VisualStudio.Services.Icons.Default)
```

![Rendered markdown screenshot](https://user-images.githubusercontent.com/11685886/147679703-c20a96e9-fe1d-4eb6-b7a2-83cdff00e412.png)

(Have included this as screenshot because GitHub Markdown does not require trailing 2 spaces for line breaks.)

**Comment by miu007** [comment]
I would also like to have this feature. I cannot find it in the current version as of today. Thanks.
